types/sequelize/index.d.ts

```diff
@@ -2908,9 +2908,9 @@ declare namespace sequelize {
         /**
          * This is the same as calling `set` and then calling `save`.
          */
-        update(key: string, value: any, options?: InstanceUpdateOptions): Promise<this>;
+        update(key: keyof TAttributes, value: any, options?: InstanceUpdateOptions): Promise<this>;
```

Comment:
You can restrict `value` here as well if you update it to `update<K extends keyof TAttributes>(key: K, value: TAttributes[K])`